featurebyte/api/target.py

```diff
@@ -67,38 +100,85 @@ def horizon(self) -> Optional[str]:
         except RecordRetrievalException:
             return self.internal_horizon
 
-    @classmethod
+    @property
+    def node(self) -> Node:
+        """
+        Representative of the current object in the graph
+
+        Returns
+        -------
+        Node
+        """
+        try:
+            node_to_return = self.cached_model.node
+        except RecordRetrievalException:
+            node_name = self.internal_node_name
+            assert node_name is not None
+            node_to_return = self.graph.get_node_by_name(node_name)
+        return node_to_return
+
+    def _get_pruned_target_model(self) -> TargetModel:
+        """
+        Get pruned model of target
+
+        Returns
+        -------
+        FeatureModel
+        """
+        pruned_graph, mapped_node = self.extract_pruned_graph_and_node()
+        target_dict = self.dict(by_alias=True)
+        target_dict["graph"] = pruned_graph.dict()
+        target_dict["node_name"] = mapped_node.name
+        return TargetModel(**target_dict)
```

Comment:
The `graph` & `node_name` of `self.dict(by_alias=True)` is already a pruned graph. Maybe we could drop the statement at line 128 (https://github.com/featurebyte/featurebyte/blob/cc24823e663e72e1a9369785d33950c637ce57cd/featurebyte/core/generic.py#L224-L236).

Reply:
hmm, there seems to be some errors if i don't explicitly call the pruning. for what it's worth, the feature class also does this explicitly pruning. i'll keep this for now and we can look to remove later if that's ok?

Reply:
Since this is only used for previewing purpose, we can keep it now. 👍